kube/src/api/dynamic.rs

```diff
@@ -188,6 +188,83 @@ impl TryFrom<DynamicResource> for Resource {
     }
 }
 
+/// Cheaply `Clone`-able string
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum StringRef {
+    /// Should be used when string is known during compilation
+    Static(&'static str),
+    /// Should be used when string is only known at runtime
+    Dynamic(Arc<str>),
+}
```

Comment:
This string wrapper does feel like a pretty complicated thing to expose in an interface. Could you explain what it's trying to optimise for? The Arc in there feels particularly confusing.

Reply:
It's purely an optimization to make cloning less expensive, especially when GVK is known at compile-time.

Do you think it should be replaced by a plain `String`?

Reply:
If it's only a handful of strings that gets cloned once, then I would worry more about the prettiness of the api more, if it's happening all the time, then it's definitely a reasonable optimization.

I think it's probably a smart thing to do internally (now that GVK has a simpler constructor), but I don't think we should expose `StringRef` to the world.

Also, curious to the tradeoff of `Arc<str>` vs just `String` inside `Dynamic`. It feels like this would just be a ref to the apps' owned String, with atomic because it's used in threads? Maybe we can use `Into<String>` in the constructor and turn it to a String  in the enum instead?

Reply:
I'll change `StringRef` to be private (as well as GVK fields) and provide two constructors (static and dynamic) instead.